File: app/src/main/java/me/devsaki/hentoid/json/GithubRelease.java
package me.devsaki.hentoid.json;

import android.webkit.MimeTypeMap;

import com.squareup.moshi.Json;

import java.util.Date;
import java.util.List;

public class GithubRelease {
    @Json(name = "tag_name")
    public String tagName;
    public String name;
    public String body;
    @Json(name = "created_at")
    public Date creationDate;
    public List<GithubAsset> assets;
    public boolean prerelease;
    public boolean draft;

    public String getName() {
        return name;
    }

    public String getBody() {
        return body;
    }

    public boolean isPublished() {
        return !prerelease && !draft;
    }

    public String getApkAssetUrl() {
        String apkMimeType = MimeTypeMap.getSingleton().getMimeTypeFromExtension("apk");
        for (GithubAsset asset : assets) {
            if (asset.content_type.equalsIgnoreCase(apkMimeType)) return asset.browser_download_url;
        }
        return "";
    }


    public static class GithubAsset {
        public String browser_download_url;
        public String content_type;
    }
}

File: app/src/main/java/me/devsaki/hentoid/json/GithubRelease.kt
package me.devsaki.hentoid.json

import android.webkit.MimeTypeMap
import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import java.util.Date

@JsonClass(generateAdapter = true)
data class GithubRelease(
    @param:Json(name = "tag_name")
    val tagName: String,
    val name: String,
    val body: String,
    @param:Json(name = "created_at")
    val creationDate: Date,
    val assets: List<GithubAsset>,
    val prerelease: Boolean,
    val draft: Boolean
) {
    fun isPublished(): Boolean {
        return !prerelease && !draft
    }

    fun getApkAssetUrl(): String {
        val apkMimeType = MimeTypeMap.getSingleton().getMimeTypeFromExtension("apk")
        for (asset in assets) {
            if (asset.contentType.equals(apkMimeType, ignoreCase = true))
                return asset.browserDownloadUrl
        }
        return ""
    }


    @JsonClass(generateAdapter = true)
    data class GithubAsset(
        @param:Json(name = "browser_download_url")
        val browserDownloadUrl: String,
        @param:Json(name = "content_type")
        val contentType: String
    )
}